### PR TITLE
feat(CX-3669): Update Select component

### PR DESCRIPTION
**Pull request by MrSltun**
This PR will be merged to `mrsltun/CX-3330/new-input-component`

This PR resolves [CX-3669]

### Description
This PR updates the Select component to match the new design
As well as removing the `variant`s from the Select component

### Screenshots
| Status | Screenshot |
| -- | -- |
| default | image |
| default (with value) | image |
| with title | image |
| with title (with value) | image |
| with title and description | image image |
| required | image |
| hover | image |
| focused | image |
| with error message | image |
| disabled | image |
| disabled (with value) | image |

### Demo Video
<video src="https://github.com/artsy/palette/assets/20655703/f080e3bc-8b1e-47ce-b40e-6d427799bed7" />

[CX-3669]: https://artsyproduct.atlassian.net/browse/CX-3669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dzucconi — APPROVED** [review]

**Comment by dzucconi** [comment]
The transition does look a bit awkward to my eye. Does the element not gain focus until something is selected?

**Review by MounirDhahri — APPROVED** [review]
LGTM

**Comment by MrSltun** [comment]
@dzucconi Yes and no, the idea behind it is, if the selected option (the first one in the list by default) has `value=""` it would show the label instead, otherwise it would show the selected option

But I understand that it's a bit weird, I'm thinking of animating the label on focus instead of when a selected option has a value

**Comment by MounirDhahri** [comment]
@MrSltun maybe we can already merge this and then create a follow-up ticket to look into the animation separately

**Comment by MrSltun** [comment]
Okay I'll merge this one and create a ticket to address the animation one :)

File: packages/palette/src/elements/Select/Select.tsx [hunk 1]
```diff
@@ -220,9 +192,34 @@ const Container = styled(Box)<ContainerProps>`
...
 `
+
+const StyledLabel = styled.label`
+  position: absolute;
+  top: 50%;
+  left: 5px;
+  padding: 0 5px;
+  pointer-events: none;
+  transform: translateY(-50%);
+  transition: 0.25s cubic-bezier(0.64, 0.05, 0.36, 1);
```
Comment (damassi) [inline packages/palette/src/elements/Select/Select.tsx:221]: The animation feels a bit unnatural on this - maybe try this setting? 

Screenshot 2023-05-11 at 8 57 00 AM

```
transition: all 500ms cubic-bezier(0.230, 1.000, 0.320, 1.000); /* easeOutQuint */
```

This is a good tool to play with animations: https://matthewlein.com/tools/ceaser

@dzucconi keeps bringing it up, but we really need some standard guidelines on animation!

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette-charts@36.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-charts%4036.0.0), [`@artsy/palette@37.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4037.0.0) :rocket:

**Pull request merged**